### PR TITLE
feat(uba): vercel `GET /uba-client-state` handler

### PR DESCRIPTION
**Pull request by dohaki**
Closes ACX-1295

Leaving as draft until testing is done

### CODE REVIEW

File: api/_utils.ts [hunk 1]
```diff
@@ -71,6 +82,21 @@ _ENABLED_ROUTES.routes = _ENABLED_ROUTES.routes.filter(
...
+  new Set(ENABLED_ROUTES.routes.map(({ fromTokenSymbol }) => fromTokenSymbol))
+);
+
+// Complete set of chain IDs ever supported by the HubPool
+export const SUPPORTED_CHAIN_IDS =
+  HUB_POOL_CHAIN_ID === 1
+    ? [
+        sdk.constants.CHAIN_IDs.MAINNET,
+        sdk.constants.CHAIN_IDs.ARBITRUM,
+        sdk.constants.CHAIN_IDs.BOBA,
```
Comment (dohaki) [inline api/_utils.ts:95]: Not sure about this one 🤔 Can this be omitted or do we need to add this

File: api/_utils.ts [hunk 2]
```diff
@@ -154,6 +180,34 @@ export const getLogger = (): LoggingUtility => {
   return logger;
 };
 
+export const getWinstonLogger = () => {
```
Comment (dohaki) [inline api/_utils.ts:183]: I had add winston logger because there were type incompatibilities with the new clients

File: api/uba-client-state.ts [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+        return spokePoolDeploymentBlock.number;
+      })
+    );
+
+    const configStoreClient = new clients.AcrossConfigStoreClient(
+      logger,
+      getAcrossConfigStore(),
+      {
+        fromBlock: HUB_POOL_DEPLOYMENT_BLOCK,
+        maxBlockLookBack: 10_000,
```
Comment (dohaki) [inline api/uba-client-state.ts:59]: Do these make sense?

File: api/uba-client-state.ts [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+    const hubPoolClient = new clients.HubPoolClient(
+      logger,
+      getHubPool(),
+      configStoreClient,
+      HUB_POOL_DEPLOYMENT_BLOCK,
+      HUB_POOL_CHAIN_ID,
+      {
+        fromBlock: HUB_POOL_DEPLOYMENT_BLOCK,
+        maxBlockLookBack: 10_000,
```
Comment (dohaki) [inline api/uba-client-state.ts:74]: Do these make sense?

File: api/uba-client-state.ts [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+          ...acc,
+          [chainId]: new clients.SpokePoolClient(
+            logger,
+            getSpokePool(chainId),
+            hubPoolClient,
+            chainId,
+            SPOKE_POOLS[chainId].deploymentBlock,
+            {
+              fromBlock: spokePoolFromBlocks[i],
+              maxBlockLookBack: 10_000,
```
Comment (dohaki) [inline api/uba-client-state.ts:90]: Do theses make sense?

File: api/uba-client-state.ts [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+        };
+      },
+      {}
+    );
+
+    const ubaState = await clients.updateUBAClient(
+      hubPoolClient,
+      spokePoolClientsMap,
+      SUPPORTED_CHAIN_IDS,
+      ENABLED_TOKEN_SYMBOLS.filter((symbol) => symbol !== "ETH"),
```
Comment (dohaki) [inline api/uba-client-state.ts:102]: Does this make sense? I was getting an error `L1 token not found for ETH`

File: api/_utils.ts [hunk 7]
```diff
@@ -39,6 +57,11 @@ export const DEFAULT_GAS_MARKUP = 0;
 
 // Don't permit HUB_POOL_CHAIN_ID=0
 export const HUB_POOL_CHAIN_ID = Number(REACT_APP_HUBPOOL_CHAINID || 1);
+export const HUB_POOL_DEPLOYMENT_BLOCK =
+  HUB_POOL_CHAIN_ID === 1 ? 14819537 : 7370152;
+
+export const CONFIG_STORE_DEPLOYMENT_BLOCK =
+  HUB_POOL_CHAIN_ID === 1 ? 14717196 : 14717196;
```
Comment (nicholaspai) [inline api/_utils.ts:64]: Why are these the same block? Should it be a goerli block as the second one?

File: api/uba-client-state.ts [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+  handleErrorCondition,
+  ENABLED_TOKEN_SYMBOLS,
+  SUPPORTED_CHAIN_IDS,
+  getWinstonLogger,
+  validTokenSymbol,
+  getHubAndSpokeClients,
+} from "./_utils";
+import { TypedVercelRequest } from "./_types";
+
+const SPOKE_POOL_FROM_BLOCK_MS_OFFSET = 24 * 60 * 60 * 1000; // 24 hours
```
Comment (nicholaspai) [inline api/uba-client-state.ts:15]: This seems reasonable offset if you're only loading one UBA bundle state. As a general rule FYI if I want to load X bundle states, I assume each bundle state spans 4 hours, so I'd want X * 4 * 2 hours of offset here. 2 is there for a buffer

File: api/uba-client-state.ts [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+      hubPoolClient,
+      spokePoolClientsMap,
+      SUPPORTED_CHAIN_IDS,
+      tokenSymbols.filter((symbol) => symbol !== "ETH"),
+      true,
+      1
+    );
+    const serializedUBAState = JSON.parse(
+      clients.serializeUBAClientState(ubaClientState)
+    );
```
Comment (nicholaspai) [inline api/uba-client-state.ts:67]: FYI I think `suggested-fees` could take this `ubaClientState`, create a new `UBAClient` and pass it in to the `update` method, and then call [`getLatestFeesForDeposit`](https://github.com/across-protocol/sdk-v2/blob/738ca4d9d19bf32caf030c18226e6a6705bd57e3/src/clients/UBAClient/UBAClientWithRefresh.ts#L70)

File: api/uba-client-state.ts [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+      // connections.
+      // Using multiple instances of `JsonRpcBatchProvider` instead makes the RPC requests more reliable.
+      {
+        useBatch: true,
+        useUncached: true,
+      }
+    );
+    await hubPoolClient.configStoreClient.update();
+
+    const ubaClientState = await clients.updateUBAClient(
```
Comment (nicholaspai) [inline api/uba-client-state.ts:57]: Can you add some timers around each of these `await` calls to figure out which is taking the longest?

**Review by nicholaspai — APPROVED** [review]
This is really easy to read code +1 

Now I think we need to optimize performance as much as we can.

**Comment by dohaki** [comment]
Closing this one in favor of https://github.com/across-protocol/frontend-v2/pull/790

**Pull request closed without merging**